saya/core/src/prover/atlantic/layout_bridge.rs

```diff
@@ -77,27 +75,26 @@ where
                 .await
             {
                 Ok(proof) => {
-                    trace!(
-                        "Proof already generated for block #{}",
-                        new_snos_proof.block_number
-                    );
                     let verifier_proof = String::from_utf8(proof).unwrap();
-                    let _: StarkProof = swiftness::parse(verifier_proof).unwrap().transform_to(); //Sanity check if the proof is valid
-
-                    info!("Proof generated for block #{}", new_snos_proof.block_number);
-
-                    let new_proof = BlockInfo {
-                        number: new_snos_proof.block_number,
-                        status: crate::storage::BlockStatus::SnosProofGenerated,
-                    };
-
-                    task_tx.send(new_proof).await.unwrap();
-                    continue;
+                    let proof = swiftness::parse(verifier_proof); //Sanity check if the proof is valid
+                    if proof.is_ok() {
+                        trace!(
+                            block_number = new_snos_proof.block_number;
+                            "Proof already generated for block"
+                        );
+                        let block_info = BlockInfo {
+                            number: new_snos_proof.block_number,
+                            status: crate::storage::BlockStatus::SnosProofGenerated,
+                        };
+
+                        task_tx.send(block_info).await.unwrap();
+                        continue;
+                    }
```

Comment:
There's an issue here, because even if it's just a sanity check, if it fails, it will be silent. Since you have now a case when the proof is valid, let's add a case to log an invalid proof format to make sure we know what's happening?

Reply:
@chudkowsky just made this slight change with some celestia changes. 🫡